src/main/java/edu/cornell/kfs/module/purap/web/filter/PurchasOrderAuthFilter.java

```diff
@@ -0,0 +1,90 @@
+package edu.cornell.kfs.module.purap.web.filter;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.HttpMethod;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import java.nio.charset.StandardCharsets;
+
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+
+import com.google.gson.Gson;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.service.WebServiceCredentialService;
+
+public class PurchasOrderAuthFilter implements Filter {
+    private static final Logger LOG = LogManager.getLogger();
+    private static final Gson gson = new Gson();
+    
+    private WebServiceCredentialService webServiceCredentialService;
+    
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        LOG.debug("doFilter, entering");
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        httpServletResponse.setHeader(CUKFSConstants.ACCESS_CONTROL_HEADER_NAME, HttpMethod.GET);
+
+        checkAuthorization(httpServletRequest, httpServletResponse, chain);
+    }
+    
+    private void checkAuthorization(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws IOException {
+        try {
+            if (isAuthorized(request)) {
+                chain.doFilter(request, response);
+            } else {
+                LOG.warn("checkAuthorization unauthorized {} {}", request.getMethod(), request.getPathInfo());
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                response.getWriter().println(gson.toJson(CUKFSConstants.UNAUTHORIZED));
+            }
+        } catch (Exception ex) {
+            LOG.error("checkAuthorization", ex);
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            response.getWriter().println(gson.toJson(CUKFSConstants.UNAUTHORIZED));
+        }
+    }
+    
+    private boolean isAuthorized(HttpServletRequest request) {
+        byte[] byteArrayEncodedAllowableUserNamePassword = Base64
+                .encodeBase64(getAllowableUserNamePassword().getBytes());
```

Comment:
You may want to use the `getBytes()` variant that accepts a charset as input.

Reply:
Thanks for the explanation. I'm fine with this remaining as-is for now, and its refactoring can wait for the shared authorization setup like you had mentioned.